homeassistant/components/lock/homematic.py

```diff
@@ -0,0 +1,59 @@
+"""
+Support for Homematic lock.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/lock.homematic/
+"""
+import logging
+from homeassistant.components.lock import LockDevice, SUPPORT_OPENING
+from homeassistant.components.homematic import HMDevice, ATTR_DISCOVER_DEVICES
+from homeassistant.const import STATE_UNKNOWN
+
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['homematic']
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Homematic lock platform."""
+    if discovery_info is None:
+        return
+
+    devices = []
+    for conf in discovery_info[ATTR_DISCOVER_DEVICES]:
+        new_device = HMLock(conf)
+        devices.append(new_device)
+
+    add_devices(devices)
+
+
+class HMLock(HMDevice, LockDevice):
+    """Representation of a Homematic lock aka KeyMatic."""
+
+    @property
+    def is_locked(self):
+        """Return true if the lock is locked."""
+        return self._hmdevice.is_locked()
```

Comment:
You can only access to `self._data` inside a property

Reply:
@pvizeli Oh okay... Can you give a short explanation why this is the case?

Reply:
@pvizeli I think this is resolved now by returning `not self._hm_get_state()` here. 

However it is possable to access `self._hmdevice`, after the homematic device is linked. 
Durring my testing the  `is_locked` method is only called after the device is linked and the `self._hmdevice` is set up.
Please correct me if I am wrong and explain it in a little more detail.